dist/aframe-altspace-component.js

```diff
@@ -465,86 +467,744 @@
-	        this.el.setAttribute('tracked-controls', 'controller', 0);
-	        this.systemGamepads = this.trackedControlsSystem.controllers.length;
-	      }
+		  if (
+			this.trackedControlsSystem &&
+			this.systemGamepads !== this.trackedControlsSystem.controllers.length &&
+			window.altspace && altspace.getGamepads && altspace.getGamepads().length
+		  ) {
+			var components = this.el.components;
+			if (components['paint-controls']) {
+			  this.gamepadIndex = components['paint-controls'].data.hand === 'left' ? 2 : 1;
+			}
+			if (this.gamepadIndex === null && components['hand-controls']) {
+			  this.gamepadIndex = components['hand-controls'].data === 'left' ? 2 : 1;
+			}
+			if (this.gamepadIndex === null && components['vive-controls']) {
+			  this.gamepadIndex = components['vive-controls'].data.hand === 'left' ? 2 : 1;
+			}
+			if (this.gamepadIndex === null && components['tracked-controls']) {
+			  this.gamepadIndex = components['tracked-controls'].data.controller;
+			}
+			this.el.setAttribute('tracked-controls', 'id', altspace.getGamepads()[this.gamepadIndex].id);
+			this.el.setAttribute('tracked-controls', 'controller', 0);
+			this.systemGamepads = this.trackedControlsSystem.controllers.length;
+		  }
 	  }
 	});
 
+	AFRAME.registerSystem('sync-system',
+	{
+		schema: {
+			author: { type: 'string', default: null },
+			app: { type: 'string', default: null },
+			instance: { type: 'string', default: null },
+			'ref-url': { type: 'string', default: null },
+		},
+		init: function() {
+			var component = this;
+
+			altspace.utilities.sync.connect({
+				authorId: this.data.author,
+				appId: this.data.app,
+				instanceId: this.data.instance,
+				baseRefUrl: this.data['ref-url']
+			}).then(function(connection) {
+				this.connection = connection;
+				this.sceneEl.emit('connected', null, false);
+
+				this.sceneRef = this.connection.instance.child('scene');
+				this.clientsRef = this.connection.instance.child('clients');
+
+				// temporary way of having unique identifiers for each client
+				this.clientId = this.sceneEl.object3D.uuid;
+				var masterClientId;
+				this.clientsRef.on("value", function (snapshot) {
+					var clientIds = snapshot.val();
+
+					var masterClientKey = Object.keys(clientIds)[0];
+					masterClientId = clientIds[masterClientKey];
+				});
+
+				this.clientsRef.on('child_added', function(childSnapshot) {
+					var joinedClientId = childSnapshot.val();
+					//let the master client flag get set first
+					setTimeout(function(){
+						component.sceneEl.emit('clientjoined', {id: joinedClientId}, false);
+					}, 1);
+				});
+
+				this.clientsRef.on('child_removed', function(childSnapshot) {
+					var leftClientId = childSnapshot.val();
+					//let the master client flag get set first
+					setTimeout(function(){
+						component.sceneEl.emit('clientleft', {id: leftClientId}, false);
+					}, 1);
```

Comment:
Are you sure? I see a quote from the old docs saying "Value events are always triggered last"